config/routes.rb

```diff
@@ -1,6 +1,12 @@
 Rails.application.routes.draw do
+  devise_for :users
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
-  # root "articles#index"
+   root "splash#index"
+  resources :main, only: %i[index]
+  resources :users
+  resources :categories, only: %i[index new create update show destroy]  do
```

Comment:
[REQUIRED] You include an update under the created resources even though you have no action for it in your controller. Kindly ensure that you are creating only resources that are used.